### PR TITLE
Add support for Beaglebone Blue

### PR DESCRIPTION
**Pull request by willnewton**
This PR adds support for the Beaglebone Blue board. I tested it with the flasher and SD card images on both a Beaglebone Blue and Black. The wifi on the Blue comes up and associates and the ethernet works on the Black but I haven't tested any other peripherals. I also haven't tested the Beaglebone Green or Green Wifi due to lack of hardware.

Instead of adding a new machine type I have added patches to the "beaglebone" machine type which should make u-boot find the correct DTB file based on the detected hardware revision so the "beaglebone" machine type should become a universal machine type eventually. I expect it to boot and run on the Green and Green Wifi but I haven't tested it.

### CODE REVIEW

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/0001-am335x_evm-uEnv.txt-bootz-n-fixes.patch [hunk 1]
```diff
@@ -0,0 +1,666 @@
+From fcc86b52d79f23b671bbe4790051ac1a6a13b7d7 Mon Sep 17 00:00:00 2001
+From: Robert Nelson <robertcnelson@gmail.com>
+Date: Tue, 14 Mar 2017 09:35:43 -0500
+Subject: [PATCH] am335x_evm: uEnv.txt, bootz, n fixes
+
+Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/0001-am335x_evm-uEnv.txt-bootz-n-fixes.patch]: Upstream status needed.

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/0002-U-Boot-BeagleBone-Cape-Manager.patch [hunk 2]
```diff
@@ -0,0 +1,744 @@
+From 385e2e1198a92d2b2932f4fb6ec4ac3d1fd51f5d Mon Sep 17 00:00:00 2001
+From: Robert Nelson <robertcnelson@gmail.com>
+Date: Sun, 5 Mar 2017 19:49:27 -0600
+Subject: [PATCH 2/2] U-Boot: BeagleBone Cape Manager
+
+Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/0002-U-Boot-BeagleBone-Cape-Manager.patch]: Upstream status needed.

**Comment by agherzan** [comment]
Add to witelist

**Comment by agherzan** [comment]
ok to test

File: layers/meta-resin-beaglebone/recipes-kernel/linux/linux-beagleboard_4.4.bb [hunk 3]
```diff
@@ -21,9 +21,9 @@ COMPATIBLE_MACHINE = "beaglebone"
 
 S = "${WORKDIR}/git"
 
-TAG = "4.4.21-ti-rt-r46"
+TAG = "4.4.55-ti-rt-r94"
```
Comment (floion) [inline layers/meta-resin-beaglebone/recipes-kernel/linux/linux-beagleboard_4.4.bb]: Is this kernel bump needed?

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/set-fdtdir-to-boot.patch [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+diff --git a/include/configs/ti_armv7_common.h b/include/configs/ti_armv7_common.h
+index 59dea72..725b6cd 100644
+--- a/include/configs/ti_armv7_common.h
++++ b/include/configs/ti_armv7_common.h
+@@ -111,7 +111,7 @@
+ 	"loadbootenv=load ${devtype} ${bootpart} ${loadaddr} ${bootenvfile}\0" \
+ 	"loadimage=load ${devtype} ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
+ 	"loadrd=load ${devtype} ${bootpart} ${rdaddr} ${bootdir}/${rdfile}; setenv rdsize ${filesize}\0" \
+-	"loadfdt=echo loading ${fdtdir}/${fdtfile} ...; load ${devtype} ${bootpart} ${fdtaddr} ${fdtdir}/${fdtfile}\0" \
++	"loadfdt=setenv fdtdir /boot; echo loading ${fdtdir}/${fdtfile} ...; load ${devtype} ${bootpart} ${fdtaddr} ${fdtdir}/${fdtfile}\0" \
```
Comment (floion) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/set-fdtdir-to-boot.patch]: how is ftdir set in the first place without your patch?

File: layers/meta-resin-beaglebone/recipes-kernel/linux/linux-beagleboard_4.4.bb (hunk 3 above)
Comment (willnewton) [inline layers/meta-resin-beaglebone/recipes-kernel/linux/linux-beagleboard_4.4.bb]: Yes, the kernel bump is needed to get the Beaglebone Blue dts included in the kernel.

**Comment by willnewton** [comment]
After talking to the upstream for the patches a set of smaller patches has been submitted to u-boot upstream. I switched to using those instead as they are much less invasive.

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/U-Boot-1-3-config-am335x_evm-detect-Black-Wireless-using-BWA.patch [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+index 8eaf3e9a5c..304ae34fb7 100644
+--- a/board/ti/am335x/board.c
++++ b/board/ti/am335x/board.c
+@@ -651,6 +651,13 @@ int board_late_init(void)
+ #ifdef CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
+ 	char *name = NULL;
+ 
++	if (board_is_bone_lt()) {
++		/* BeagleBoard.org BeagleBone Black Wireless: */
++		if (!strncmp(board_ti_get_rev(), "BWA", 3)) {
```
Comment (michal-mazurek) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/U-Boot-1-3-config-am335x_evm-detect-Black-Wireless-using-BWA.patch]: The strncmp only checks the prefix of board_ti_get_rev(), not the entire string, against "BWA". I wonder if this is deliberate or a logical error that might bite us in the future.

**Comment by agherzan** [comment]
Re-reviewed - looks great.

**Pull request merged**